mantle/cmd/kola/testiso.go

```diff
@@ -190,41 +230,144 @@ func testLegacyInstaller() error {
 		if err := cmd.Run(); err != nil {
 			return errors.Wrapf(err, "running gzip")
 		}
+	} else {
+		if err := absSymlink(filepath.Join(builddir, metalimg), filepath.Join(tftpdir, metalimg)); err != nil {
+			return err
+		}
+	}
+
+	for _, name := range []string{kern.kernel, kern.initramfs} {
+		if err := absSymlink(filepath.Join(builddir, name), filepath.Join(tftpdir, name)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func setupTest(kern *kernelSetup) (*installerTest, error) {
+	builder := platform.NewBuilder(kola.QEMUOptions.Board, "")
+	builder.Firmware = kola.QEMUOptions.Firmware
+	builder.AddDisk(&platform.Disk{
+		Size: "12G", // Arbitrary
+	})
+
+	// This applies just in the legacy case
+	builder.Memory = 1542
+	if kola.QEMUOptions.Board == "s390x-usr" {
+		// FIXME - determine why this is
+		builder.Memory = int(math.Max(float64(builder.Memory), 16384))
+	}
+
+	// For now, but in the future we should rely on log capture
+	builder.InheritConsole = true
+
+	tempdir, err := ioutil.TempDir("", "kola-testiso")
+	if err != nil {
+		return nil, err
+	}
+	cleanupTempdir := true
+	defer func() {
+		if cleanupTempdir {
+			os.RemoveAll(tempdir)
+		}
+	}()
+
+	tftpdir := filepath.Join(tempdir, "tftp")
+	if err := os.Mkdir(tftpdir, 0777); err != nil {
+		return nil, err
+	}
+
+	builddir := filepath.Dir(kola.Options.CosaBuild)
+	metalimg := kola.CosaBuild.BuildArtifacts.Metal.Path
+	metalname := metalimg
+	// Yeah this is duplicated with setupTftpDir
+	if strings.HasSuffix(metalimg, ".raw") {
+		metalname = metalimg + ".gz"
+	}
+
+	if err := setupTftpDir(builddir, tftpdir, metalimg, kern); err != nil {
+		return nil, err
+	}
+
+	pxe := pxeSetup{}
+	pxe.tftpipaddr = "192.168.76.2"
+	switch kola.QEMUOptions.Board {
+	case "amd64-usr":
+		pxe.boottype = "pxe"
+		pxe.networkdevice = "e1000"
+		pxe.pxeimagepath = "/usr/share/syslinux/"
+		break
+	case "ppc64le-usr":
+		pxe.boottype = "grub"
+		pxe.networkdevice = "virtio-net-pci"
+		break
+	case "s390x-usr":
+		pxe.boottype = "pxe"
+		pxe.networkdevice = "virtio-net-ccw"
+		pxe.tftpipaddr = "10.0.2.2"
+		pxe.bootindex = "1"
+	default:
+		return nil, fmt.Errorf("Unsupported arch %s", kola.QEMUOptions.Board)
 	}
 
 	mux := http.NewServeMux()
 	mux.Handle("/", http.FileServer(http.Dir(tftpdir)))
 	listener, err := net.Listen("tcp", ":0")
 	if err != nil {
-		return err
+		return nil, err
 	}
 	port := listener.Addr().(*net.TCPAddr).Port
+	// Yeah this leaks
```

Comment:
Orly?

Reply:
Yeah, need to do something like hook into the `destroy()` and call `http.Shutdown`.  But...later?